### PR TITLE
Feature/smoke test and application test

### PR DESCRIPTION
**Pull request by KareemEH**

### CODE REVIEW

File: .github/workflows/maven-tests.yml [hunk 1]
```diff
@@ -6,12 +6,30 @@ jobs:
   build:
     runs-on: ubuntu-latest
 
+    services:
```
Comment (khalidkanaan) [inline .github/workflows/maven-tests.yml]: This is perfect, we need the same container as the docker-compose to ensure testing is done properly

**Review by khalidkanaan — APPROVED** [review]
Looks good.

File: .github/workflows/maven-tests.yml (hunk 1 above)
Comment (KareemEH) [reply .github/workflows/maven-tests.yml]: Addressed.

**Review by khalidkanaan — APPROVED** [review]
Good

**Pull request merged**